daisy/workflow/test_common.go

```diff
@@ -71,15 +71,15 @@ func init() {
 }
 
 func testWorkflow() *Workflow {
-	ctx, cancel := context.WithCancel(context.Background())
+	cancel := make(chan struct{})
```

Comment:
For canceling the running workflow it works fine, the issue is we also use this context for cleanup which happens after a cancel call.